content/general/concepts/creational-design-patterns/creational-design-patterns.md

```diff
@@ -0,0 +1,23 @@
+---
+Title: 'Creational Design Patterns'
+Description: 'Concern themselves with the instantiation, or creation, of objects.'
+Subjects:
+  - 'Computer Science'
+  - 'Interview Prep'
+Tags:
+  - 'Classes'
+  - 'Conceptual'
+  - 'Constructors'
+  - 'Objects'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+Creational design patterns concern themselves with the instantiation of objects. At time of creation, additional logic may be needed to decide what or how object type needs to be created. There are five main creational design patterns, outlined below:
+
+- [Singleton](https://www.codecademy.com/resources/docs/general/singleton-pattern): Ensures a class has only one object instantiated during the programs lifecycle.
+- [Builder](https://www.codecademy.com/resources/docs/general/builder-pattern): Decouples the responsibility of object creation from the desired object's class. It is normally implemented when a class requires many properties to be set during its construction.
+- [Prototype](https://www.codecademy.com/resources/docs/general/prototype-pattern): Adds cloning functionality to an objects class.
+- [Factory](https://www.codecademy.com/resources/docs/general/factory-pattern): Creates objects without specifying the specific class.
+- [Abstract Factory](https://www.codecademy.com/resources/docs/general/abstract-factory-pattern): Groups object factories based on a common theme.
```

Comment:
@yangc95 Yes sure. I get emails from GitHub, so if you tag me in anything, I'd be more than happy to look / help where I can :)

Just a little information about design patterns (as it may effect how you are thinking of things.) There was a book written some time ago titled Design Patterns (re-useable software) or something like that, in which the author described 3 ways common bits of software behave. Creational was one of the 3. For the books code e.g. I think it was written in C, but not sure. Just to let you know I have never read the book.

Below would show you how all the patterns are structured (by titles) and the idea would be to get Docs to have something written about each of them. Please bare in mind this is quite a large piece of work. As a back end java developer, if you have knowledge of the patterns, understand APIs (SOAP / REST) and MQ for communication, and understand how to test (unit, integration, end to end ...) then I think you'd have a pretty good idea of how to break down the back end into manageable chunks and pass the fount useful / appropriate data. My hope was that we can get some of this more complex info into Codecademy somehow, to allow the site to become more useful to those who may have found employment, and are undertaking post introductory training.

- Design Patterns
  - Creational
    - Singleton
    - Builder
    - Prototype
    - Factory
    - Abstract Factory
  - Behavioural
    - Chain of Responsibility
    - Command
    - Interpreter
    - Iterator
    - Mediator
    - Memento
    - Observer
    - State
    - Strategy
    - Template
    - Visitor
  - Structural 
    - Adaptor
    - Bridge
    - Composite
    - Decorator
    - Facade
    - Flyweight
    - Proxy
    
Hope that helps explain what I'm kind of aiming towards with these design pattern entries :)

Reply:
@Bernard-McGeever Cool, I think I get the idea. Definitely going to in with the mindset of keeping the level as intended for intermediate learners while trying to bridge the gap for beginners, or at least inform learners on what is required prior to these patterns.

Thanks for all those pointers 😄 I'll take a dive into the book you've provided and other resources along the way. Will update you accordingly!